src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure(int mode) {
+        Figure figure;
+        switch(mode) {
+            case 0: figure = new Circle(random.nextDouble());

```

Comment:
Avoid using magic numbers directly in your code. Consider defining a constant for the number of figure types, as suggested in the checklist: `public static final int FIGURE_COUNT = 5;` and use it in the `switch` statement.